packages/msgraph/test/Adaptor.test.js

```diff
@@ -38,6 +38,31 @@ describe('execute', () => {
       expect(finalState).to.eql({ references: [], data: null });
     });
   });
+
+  it('should stop operation on error', async () => {
+    const state = {
+      drives: {
+        default: {
+          id: 'b!YXzpkoLwR06bxC8tNdg71m_',
+        },
+      },
+    };
+    const operations = [
+      state => {
+        state.counter++;
+        throw new Error('Failed operation');
+      },
+      state => {
+        return { ...state, counter: 1 };
+      },
+    ];
+
+    const finalState = await execute(...operations)(state).catch(e => {
+      expect(e.message).to.contain('Failed operation');
```

Comment:
@mtuchi the fix looks fine but if I comment it out, your test still passes :grimacing: 

If execute doesn't throw, this assertion doesn't run, and so the test passes.

You need to do something like this (this is untested):
```
    let e;
    const finalState = await execute(...operations)(state).catch(err => {
      e = err;
    });

    expect(e.message).to.contain('Failed operation');
```

You probably need to assert that e exists (ie, the catch handler was called) and that it has the right value

Reply:
Done [cb6adb0](https://github.com/OpenFn/adaptors/pull/318/commits/cb6adb0a0ac66c48f3e66666217edce4bcd61cc0)